go/arrow/array/data_test.go

```diff
@@ -49,3 +49,46 @@ func TestDataReset(t *testing.T) {
 		data.Reset(&arrow.Int64Type{}, 5, data.Buffers(), nil, 1, 2)
 	}
 }
+
+func TestSizeInBytes(t *testing.T) {
```

Comment:
we should add a test for a sliced array, a sliced array with children, and an array with children which are slices. 

As per my comment above, it should confirm that `SizeInBytes` returns values that are upper bounds, not taking the offset into account.

Reply:
Addressed.